### PR TITLE
Support garbage collection after pt2 compilation

### PR DESCRIPTION
**Pull request by qiurc**
Summary: 
Support garbage collection after pt2 compilation.
Add jk to control the global rollout / rollback of this functionality
Add env var to control individual job's rollout

Test Plan:
Test the model training job with / without this changes

Reviewers:
@yuxihu @ezyang , @Yuzhen11 , 

Subscribers:

Tasks:

Tags:

Fixes #ISSUE_NUMBER

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

### CODE REVIEW

File: torch/_dynamo/convert_frame.py [hunk 1]
```diff
@@ -349,9 +350,9 @@ def exception_handler(
 
 
 FRAME_COUNTER = 0
-FRAME_COMPILE_COUNTER: typing.Counter[
-    Union[int, FrameStateSizeEntry]
-] = collections.Counter()
+FRAME_COMPILE_COUNTER: typing.Counter[Union[int, FrameStateSizeEntry]] = (
+    collections.Counter()
+)
```
Comment (ezyang) [inline torch/_dynamo/convert_frame.py:355]: fbcode formatter causing thrash with the oss formatter. you may fail lint this way

**Review by ezyang — APPROVED** [review]
But you may want to undo the unrelated formatting changes

File: torch/_dynamo/convert_frame.py (hunk 1 above)
Comment (qiurc) [reply torch/_dynamo/convert_frame.py:355]: Thanks for pointing it out! Let me revert them.

**Pull request closed without merging**